src/checkers/inference/solver/backend/maxsat/encoder/MaxSATComparableConstraintEncoder.java

```diff
@@ -39,19 +39,42 @@ public MaxSATComparableConstraintEncoder(Lattice lattice, ConstraintVerifier ver
         return result;
     }
 
-    // TODO I'm not sure why in the original GTIS implementation, the below three methods returns emptyValue?
     @Override
     public VecInt[] encodeVariable_Constant(VariableSlot fst, ConstantSlot snd) {
-        return emptyValue;
+        if (lattice.incomparableType.keySet().contains(snd.getValue())) {
+            List<VecInt> resultList = new ArrayList<>();
+            for (AnnotationMirror incomparable : lattice.incomparableType.get(snd.getValue())) {
+                // Should not be equal to incomparable
+                resultList.add(
+                    VectorUtils.asVec(
+                        -MathUtils.mapIdToMatrixEntry(fst.getId(), typeToInt.get(incomparable), lattice)));
+            }
+            VecInt[] resultArray = new VecInt[resultList.size()];
+            return resultList.toArray(resultArray);
+        } else {
+            return emptyValue;
+        }
     }
 
     @Override
     public VecInt[] encodeConstant_Variable(ConstantSlot fst, VariableSlot snd) {
-        return emptyValue;
+        if (lattice.incomparableType.keySet().contains(fst.getValue())) {
```

Comment:
Instead of duplicating this logic, can you please extract this in a helper function - or, as Comparable is symmetric, you could just call `encodeVariable_Constant` with swapped parameters.
(Should this in general be the implementation for symmetric operations like comparable and equality? Are const-var vs var-const ever interesting?)